src/chat/Chat.ts

```diff
@@ -0,0 +1,161 @@
+import { socket } from '../app'
+import { format } from 'date-fns'
+import $ from 'jquery'
+import { addMessage, addMessagesToRoom } from '../messages'
+import {
+    GetAllMessagesResponse,
+    ImageMessage,
+    MessageType,
+    Sockets,
+    TextMessage
+} from '../types'
+
+let allMessages: Array<ImageMessage | TextMessage> = []
+export const hours = format(new Date(), 'HH:mm')
+
+export const createChatMessage = () => {
+    const windowMessage = $('.message')
+    const inputText = $('.input--text')
+
+    socket.on(Sockets.RoomMessage, (roomMessage: string, allMessageInRoom: GetAllMessagesResponse) => {
+        if (allMessageInRoom) {
+            const allMessage = allMessageInRoom.allMessages
```

Comment:
you can use destructuring

Reply:
I can't use destructing at this object, because it isn't iterable, please look at the type

export interface GetAllMessagesResponse {
    allMessages: Array<ImageMessage | TextMessage>
}

Reply:
```
const { allMessages } = allMessagesInRoom
```